pkg/stanza/fileconsumer/file.go

```diff
@@ -63,7 +55,7 @@ func (m *Manager) Start(persister operator.Persister) error {
 		if len(offsets) > 0 {
 			m.Infow("Resuming from previously known offset(s). 'start_at' setting is not applicable.")
 			m.readerFactory.FromBeginning = true
-			m.knownFiles = append(m.knownFiles, offsets...)
+			m.knownFiles[1].Add(offsets...)
```

Comment:
Well, if we do knownFiles[0] and one of the known files disappears due to some reasons and is again detected after a poll() then it will cause re-reading of all contents.
But yeah, this can also be done as knownFiles[0] instead of knownFiles[1]. I just took a safeguard measure.